weld/sir.rs

```diff
@@ -654,15 +656,15 @@ fn sir_param_correction(prog: &mut SirProgram) -> WeldResult<()> {
 }
 
 /// Convert an AST to a SIR program. Symbols must be unique in expr.
-pub fn ast_to_sir(expr: &TypedExpr) -> WeldResult<SirProgram> {
+pub fn ast_to_sir(expr: &TypedExpr, multithreaded: bool) -> WeldResult<SirProgram> {
```

Comment:
Added this to make it easier to do single vs. multithreaded optimizations in the SIR later.